bittensor/_receptor/receptor_impl.py

```diff
@@ -123,20 +123,32 @@ def __del__ ( self ):
     def __exit__ ( self ):
         self.__del__()
 
-    def sign ( self ):
+    def sign_v1( self ):
         r""" Uses the wallet pubkey to sign a message containing the pubkey and the time
         """
-        nounce = self.nounce()
-        message  = str(nounce) + str(self.wallet.hotkey.ss58_address) + str(self.receptor_uid)
+        nonce = self.nonce()
+        message  = str(nonce) + str(self.wallet.hotkey.ss58_address) + str(self.receptor_uid)
         spliter = 'bitxx'
-        signature = spliter.join([ str(nounce), str(self.wallet.hotkey.ss58_address), "0x" + self.wallet.hotkey.sign(message).hex(), str(self.receptor_uid) ])
+        signature = spliter.join([ str(nonce), str(self.wallet.hotkey.ss58_address), "0x" + self.wallet.hotkey.sign(message).hex(), str(self.receptor_uid) ])
         return signature
-    
-    def nounce ( self ):
+
+    def sign_v2(self):
+        nonce = f"{self.nonce()}"
+        sender_hotkey = self.wallet.hotkey.ss58_address
+        receiver_hotkey = self.endpoint.hotkey
+        message = f"{nonce}.{sender_hotkey}.{receiver_hotkey}.{self.receptor_uid}"
+        signature = f"0x{self.wallet.hotkey.sign(message).hex()}"
+        return ".".join([nonce, sender_hotkey, signature, self.receptor_uid])
+
+    def sign(self):
```

Comment:
I am missing unit tests for this function, can you add them?

Reply:
So this should be covered by `test_axon_receptor_connection_forward_works` - it varies between the old endpoint version (which uses `sign_v1`), new endpoint version (which uses `sign_v2`) and the current version (just future proofing).

It looks like I accidentally broke the test name while adding this matrix test - should be fixed now.

For that matter - do you have a line coverage overview somewhere ? I couldn't find it in the CircleCI runs.

Reply:
commented offline,

- no coverage available right now.
- a dedicated unit test for this will help us localize the failure if that happens and ensure that we have the code covered by a tests. Unit tests coverage should always be specific pieces of code that cover the code we added and not a side effect of a bigger test